### PR TITLE
Rewrite WelcomePage

### PR DESCRIPTION
**Pull request by danrabbit**
* Inline namespace
* GObject style
* Make sure we create buttons before we try to change their visibility by adding signals at the end
* Reduce variable scope where possible
* Update replay button title and subtitle when `last-stopped` changes
* Update replay button visibility and launch the correct file when `current-video` changes
* DRY

### CODE REVIEW

File: src/Widgets/WelcomePage.vala [hunk 1]
```diff
@@ -15,103 +15,101 @@
...
+        );
+    }
 
-            if (settings.get_double ("last-stopped") == 0.0 || !settings.get_boolean ("resume-videos")) {
-                append ("media-playlist-repeat", _("Replay last video"), get_title (last_file.get_basename ()));
-            } else {
-                append ("media-playback-start", _("Resume last video"), get_title (last_file.get_basename ()));
+    construct {
+        append ("document-open", _("Open file"), _("Open a saved file."));
+        append ("media-playlist-repeat", _("Replay last video"), "");
```
Comment (jeremypw) [inline src/Widgets/WelcomePage.vala:31]: Need to initialise the label of the button correctly according to the "last-stopped" setting. See issue #187. This issue exists in master but might as well fix it now.

Comment (danrabbit) [reply src/Widgets/WelcomePage.vala:31]: It gets set to the correct value on line #62. I guess these could just be empty strings really

Comment (danrabbit) [reply src/Widgets/WelcomePage.vala:31]: I've moved things up to make that more clear. Let me know if you think it needs a comment still

**Comment by jeremypw** [comment]
It still seems to suffer the issue in #187. Are you not getting that issue? The problem seems to be that the "last-stopped" setting changes when the window is destroyed from "0" to "0.002..." for some reason.  A quick fix would be to change the test in `update_replay_title ()` from`> 0` to `> 0.003` or change the test in the destroy callback.  Note that this issue only seems to occur with_short_ videos (e.g. 1 minute)

**Comment by danrabbit** [comment]
I'm not experiencing and issue with the time save being set to a non-zero value on destroy.

I think this sounds like a bug in a different part of the codebase though and I'm hesitant to add a workaround here instead of solving it at the source

**Comment by jeremypw** [comment]
OK - it can be left for another PR then.

**Pull request merged**